pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, pkgs
+, runtimeShell
+, symlinkJoin
+, tree
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
+      in
+        let p = callPackage datum.expressionPath parameters;
+        in acc // { ${datum.name} = p; };
+    in builtins.foldl' _packageWithLinkCommandFrom {} _wordlistData
+  );
+
+in let
+  packageConstructor = wordlistPackages:
+    symlinkJoin rec {
+      pname = "security-wordlists";
+      version = "unstable-2020-11-23";
+
+      name = "${pname}-${version}";
+
+      paths = wordlistPackages;
+
+      dontUnpack = true;
+
+      propagatedBuildInputs = [ tree ];
+
+      postBuild = ''
+        # Create a command to show the location of the links.
+        mkdir -p $out/bin
+        cat >> $out/bin/wordlists << __EOF__
+          #!${runtimeShell}
+          ${tree}/bin/tree $out/share
+        __EOF__
+        chmod +x $out/bin/wordlists
+
+        # Create a handy command for easy access to the wordlists.
+        # e.g.: `cat "$(wordlists_path)/rockyou.txt"`, or `ls "$(wordlists_path)/dirbuster"`
+        cat >> $out/bin/wordlists_path << __EOF__
+          #!${runtimeShell}
+          echo $out/share
+        __EOF__
+        chmod +x $out/bin/wordlists_path
+      '';
+
+      meta = with lib; {
+        description = "A collection of wordlists useful for security testing";
+        maintainers = with maintainers; [ pamplemousse ];
+      };
+    };
+in {
+  pkgs = scopedWordlists;
```

Comment:
Sorry for pro-longing this PR, but once it gets merged, then I would like avoid breaking people who use it if we decide to pivot.

I think this would create paths like `security-wordlists.pkgs.wfuzz`. Wouldn't it we want a scope more like `security-wordlists.wfuzz`... I guess if we have more abstractions at the `security-wordlists` package set level then maybe it would warrant another attr (e.g. `security-wordlists.all`, etc).

Other thing, is that maybe we should call it `lists` instead of `pkgs`? `pkgs` sound more like binaries/libraries. At least in my head.

I don't feel super strong about this, just want to make sure it's what we would like moving forward and "written into stone"

Reply:
> I think this would create paths like `security-wordlists.pkgs.wfuzz`

Yes, it does.

> Wouldn't it we want a scope more like `security-wordlists.wfuzz`...

Not sure that would fit the use cases highlighted in https://github.com/Pamplemousse/nixpkgs/blob/wordlists/doc/builders/packages/security-wordlists.section.md#usage ...

And AFAIR, I wanted it to have some consistency with other parts of `nixpkgs`;
In particular, being familiar `python`, I was aiming to "replicate" its structure:
  * `python` --> `security-wordlists`, a package. The latter builds two `bin/`: `wordlists` and `wordlists_path`;
  * `python.pkgs` --> `security-wordlists.pkgs`, a set of "sub-packages". The latter being the lists per se.

> Other thing, is that maybe we should call it lists instead of pkgs? pkgs sound more like binaries/libraries.

Why not. But again, what about consistency across `nixpkgs` (https://github.com/FRidh/rfcs/blob/83/rfcs/0083-common-interface-package-sets.md)?
And in the end, a wordlist is a package (see for example to import in an expression: https://github.com/Pamplemousse/nixpkgs/blob/wordlists/doc/builders/packages/security-wordlists.section.md#in-an-expression)...